File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
def solution(n, a, b):
    answer = 1
    s = n
    f = 0
    if a > b :
        a, b = b, a
        
    while s != 1 :
        s //= 2
        f += 1
    
    mid = n // 2
    left = 1
    right = n
    c = 1
    while True :
        if a <= mid and b > mid :
            break
        
        elif a <= mid and b <= mid :
            right = mid
            mid = (left + right) // 2
        
        elif a > mid and b > mid :
            left = mid + 1
            mid = (left + right) // 2
        
        c += 1
    
    answer = f - c + 1
    return answer

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution {
    public static int solution(int n, int a, int b) {
        int target1 = a - 1;
        int target2 = b - 1;

        if (big(a, b)) {
            target1 = b - 1;
            target2 = a - 1;
        }
        int answer = 1;

        while (true) {
            if (target1 + 1 == target2) {
                if (target1 / 2 == target2 / 2) {
                    break;
                }
            }
            target1 = target1 / 2;
            target2 = target2 / 2;
            answer++;
        }

        return answer;
    }

    public static boolean big(int a, int b) {
        if (a > b) {
            return true;
        }
        return false;
    }
}